keel-docker/src/main/kotlin/com/netflix/spinnaker/keel/docker/DockerImageResolver.kt

```diff
@@ -72,7 +72,7 @@ abstract class DockerImageResolver<T : ResourceSpec>(
 
   override fun invoke(resource: Resource<T>): Resource<T> {
     val container = getContainerFromSpec(resource)
-    if (container is DigestProvider) {
+    if (container is DigestProvider || container is NullableReferenceProvider) {
```

Comment:
IIUC, this comes into play in your plugin because your `ResourceSpec` has a `ContainerProvider` field (which might need to be empty if the K8s resources within don't care about a container), correct?

I was remembering that you had the opposite example in your proposal, where you might have, say, 2 different `Deployment` objects within the same MD resource spec, and would need a way to reference 2 different artifacts. That got me wondering whether perhaps a better approach might be to create a new sub-type of `ContainerProvider` that can handle multiple references (`MultiReferenceContainerProvider`?), which might look something like this:

```yaml
kind: k8s/resources@v1
spec:
  container:
    references:
      - my-first-image
      - my-second-image
   # omitted for brevity
```

...which then would also allow you do this when you don't care about containers, instead of omitting the property altogether:
```yaml
kind: k8s/resources@v1
spec:
  container:
    references: []
   # omitted for brevity
```

Thoughts?

Reply:
hmmm, yeah! love the idea of combining the two into one type. Had that multi-reference idea somehow parked when I did this. will update the PR. thanks for bringing it to my attention :)